src/main/java/graphql/introspection/IntrospectionQueryBuilder.java

```diff
@@ -170,187 +180,210 @@ public Options inputValueDeprecation(boolean flag) {
-                            ) )
-                            .selectionSet( SelectionSet.newSelectionSet()
-                                .selection( FragmentSpread.newFragmentSpread("InputValue").build() )
-                                .build()
-                            )
-                            .build(),
-                        Field.newField("type", SelectionSet.newSelectionSet()
-                                .selection( FragmentSpread.newFragmentSpread("TypeRef").build() )
+        SelectionSet fullTypeSelectionSet = newSelectionSet().selections(filter(
+                newField("kind").build(),
+                newField("name").build(),
+                options.descriptions ? newField("description").build() : null,
+                options.specifiedByUrl ? newField("specifiedByURL").build() : null,
+                newField("fields")
+                        .arguments(ImmutableList.of(
+                                newArgument("includeDeprecated", BooleanValue.of(true)).build()
+                        ))
+                        .selectionSet(newSelectionSet().selections(filter(
+                                        newField("name").build(),
+                                        options.descriptions ? newField("description").build() : null,
+                                        newField("args")
+                                                .arguments(filter(
+                                                        options.inputValueDeprecation ? newArgument("includeDeprecated", BooleanValue.of(true)).build() : null
+                                                ))
+                                                .selectionSet(newSelectionSet()
+                                                        .selection(newFragmentSpread("InputValue").build())
+                                                        .build()
+                                                )
+                                                .build(),
+                                        newField("type", newSelectionSet()
+                                                .selection(newFragmentSpread("TypeRef").build())
+                                                .build()
+                                        )
+                                                .build(),
+                                        newField("isDeprecated").build(),
+                                        newField("deprecationReason").build()
+                                )).build()
+                        )
+                        .build(),
+                newField("inputFields")
+                        .arguments(filter(
+                                options.inputValueDeprecation ? newArgument("includeDeprecated", BooleanValue.of(true)).build() : null
+                        ))
+                        .selectionSet(newSelectionSet()
+                                .selection(newFragmentSpread("InputValue").build())
```

Comment:
This looks weird in diff - but all i did here was static import classes.  The diff is having trouble here but its the same code